packages/adapter-cloudflare-workers/index.js

```diff
@@ -86,6 +94,24 @@ function validate_config(builder) {
 			);
 		}
 
+		// @ts-ignore
+		const main_file = wrangler_config.build?.upload?.main;
```

Comment:
is this needed or could/should we hardcode/use the filename for main instead? 

The checks below feel a bit brittle, what if there is no glob but an excact matching rule instead?

Reply:
May be a bug with wrangler, but main seems to be treated according to the rules, regardless of the specified `format`. My build kept failing to upload until I changed the name to .mjs or added a rule to treat *.js as ESM. `build.upload.main` is actually required, so we can probably get rid of all the `?.` there.